### PR TITLE
Add maintainers best practice document and update maintainer list

### PR DESCRIPTION
**Pull request by daxian-dbw**
## PR Summary

Add document about best practices for maintainers.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: docs/maintainers/best-practice.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  make sure you run through the following steps:
+  1. The commit title should be a concise summary of the PR.
+     - When merging with the `Create a merge commit` option,
+       the default commit title would be `Merge pull request XXX from XXX`.
+       **Replace it with a concise summary of the PR**, and add the PR number to the end, like `(#1234)`.
+     - When merging with the `Squash and merge` option,
+       the PR title will be used as the commit title by default.
+       **Reword the title as needed** to make sure it makes sense (can be used without change in `CHANGELOG.md`).
+  1. The commit description is required for feature PRs or PRs with breaking changes.
+     For other PRs, it's not required but good to have based on the judgement of the maintainer.
```
Comment (iSazonov) [inline docs/maintainers/best-practice.md]: What is feature PRs?

File: docs/maintainers/best-practice.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+     - If a PR introduces breaking changes,
+       make sure you put the tag `[breaking change]` at the first line of the description,
```
Comment (iSazonov) [inline docs/maintainers/best-practice.md]: Header or description? Git consider first line of a commit description as a header. GitHub split and show input field for commit header (first line of the commit description) and second field for commit description (starting with second line of the commit description).

Comment (daxian-dbw) [reply docs/maintainers/best-practice.md]: Experimenting with images to see if that can make it clear about `Commit Title` and `Commit Description`. But images don't look very good here. May change again, stay tuned.

**Review by iSazonov — APPROVED** [review]

**Comment by daxian-dbw** [comment]
@iSazonov I think it's much more clear and also look better now. Please take another look.

File: docs/maintainers/README.md [hunk 3]
```diff
@@ -20,14 +20,10 @@ They have [write access](https://help.github.com/articles/repository-permission-
...
 
-* Sergei Vorobev ([vors](https://github.com/vors))
-* Jason Shirk ([lzybkr](https://github.com/lzybkr))
-* Dongbo Wang ([daxian-dbw](https://github.com/daxian-dbw))
-* Travis Plunk ([TravisEz13](https://github.com/TravisEz13))
-* Mike Richmond ([mirichmo](https://github.com/mirichmo))
-* Andy Schwartzmeyer ([andschwa](https://github.com/andschwa))
-* Aditya Patwardhan ([adityapatwardhan](https://github.com/adityapatwardhan))
-* Ilya Sazonov ([iSazonov](https://github.com/iSazonov))
+- Dongbo Wang ([daxian-dbw](https://github.com/daxian-dbw))
```
Comment (anmenaga) [inline docs/maintainers/README.md]: Ironically, in this case PR Summary does not entirely correctly summaries the changes. :)
PR Summary += "... and updating maintainer list"

File: docs/maintainers/best-practice.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+## PR Types
+
+- Feature-work PR: A PR that implements an RFC, which usually involves relatively large set of changes.
+- Regular PR: A bug fix or an enhancement change that are not backed by an RFC.
+
+## Review PRs
+
+- Ask the author to reword the PR title based on guidelines in [Contributing](../../.github/CONTRIBUTING.md).
+- Ask the author to apply `[Feature]` tag to trigger full test builds if it's necessary.
+- Label the PR with `Breaking-Change` and/or `Documentation Needed` as appropriate.
```
Comment (anmenaga) [inline docs/maintainers/best-practice.md]: I would also add "... and add PR classification using other Labels if they are missing".

**Review by anmenaga — APPROVED** [review]
LGTM with minor comments.

**Review by TravisEz13 — APPROVED** [review]

File: docs/maintainers/best-practice.md (hunk 4 above)
Comment (daxian-dbw) [reply docs/maintainers/best-practice.md]: Good point. Will update.

**Review by dantraMSFT — APPROVED** [review]

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**